tests/unit/search/test_search_query.py

```diff
@@ -807,11 +807,19 @@ def test_find_report():
     assert expected == result
 
 
-@pytest.mark.xfail(reason='tracked in issue #817')
 def test_find_type_code():
     query = InspireQuery('find tc book')
 
-    expected = {}
+    expected = {'query': {'multi_match': {'query': 'book', 'fields': ['collection']}}}
+    result = query.body
+
+    assert expected == result
+
+
+def test_type_code_colon():
+    query = InspireQuery('tc: l')
+
+    expected = {'query': {'multi_match': {'query': 'l', 'fields': ['collection']}}}
     result = query.body
 
     assert expected == result
```

Comment:
As I anticipated, my test ordering isn't easily grokkable :dancer:

Shall we try after lunch try to come up with a better ordering together?